system-contracts/contracts/EvmInterpreter.sol

```diff
@@ -374,15 +375,81 @@ contract EvmInterpreter {
         }
     }
 
+    function systemCall(
```

Comment:
to reuse more code, could you please move this function to the SystemContractCaller and make it reuse logic from this function?

basically we'll have to get `dataStart/dataLength` from the `bytes memory` param and call the same function as here

Reply:
I added an explicit returndata forwarding to systemCall in a latter commit, not sure how to do it implicitly